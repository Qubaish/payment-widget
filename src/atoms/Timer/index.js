import React, { useState, useEffect, Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledTimer = styled.span`
    color: #307EF7;
    align-self: flex-end;
    padding: 5px;
`;

const Link = styled.button`
    color: #307EF7;
    background: none;
    border: none;
    pointer: cursor;
    align-self: flex-end;
    padding: 5px;
    cursor: pointer;
    :focus {
        outline: none;
    }
`;

function Timer({seconds}) {
    const [counter, setCounter] = useState(seconds);
  
    useEffect(() => {
      const timer =
        counter > 0 && setInterval(() => setCounter(counter - 1), 1000);
      return () => clearInterval(timer);
    }, [counter]);

    const resetTimer = () => setCounter(seconds);
  
    return (
        <Fragment>
            {counter > 0 && (<StyledTimer data-testid="timer">00:{counter}</StyledTimer>)}
            {counter === 0 && (<Link onClick={resetTimer} data-testid="btn">reset</Link>)}
        </Fragment>
    );
}

Timer.propTypes = {
    seconds: PropTypes.number,
}

Timer.defaultProps = {
    seconds: 10,
}

export default Timer;
