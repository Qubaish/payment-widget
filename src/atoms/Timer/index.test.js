import React from 'react';
import { render } from '@testing-library/react';
import Timer from './index';

describe('<Timer />', () => {
    it('checking snapshot', () => {
        const { asFragment } = render(<Timer seconds={2}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render a text', () => {
        const { getByTestId} = render(<Timer seconds={5}/>);
        const elem = getByTestId('timer');
        expect(elem).toBeTruthy();
    });
    it('should check after timeout show reset btn', () => {
        const { getByTestId} = render(<Timer seconds={1}/>);
        setTimeout(() => {
            const elem = getByTestId('btn');
            expect(elem.innerHTML).toBe('reset');
            expect(component.state().fruits).toEqual(fruits);
        }, 1000);
    });
});