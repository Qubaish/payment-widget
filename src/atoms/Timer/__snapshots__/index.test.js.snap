// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> checking snapshot 1`] = `
<DocumentFragment>
  <span
    class="sc-AxjAm fivlzj"
    data-testid="timer"
  >
    00:2
  </span>
</DocumentFragment>
`;
